src/stdlib/builtin/lv_string_builtin.c

```diff
@@ -232,6 +232,18 @@ int lv_strcmp(const char * s1, const char * s2)
     return *(const unsigned char *)s1 - *(const unsigned char *)s2;
 }
 
+int lv_strncmp(const char * s1, const char * s2, size_t len)
+{
+    while(len > 0 && *s1 && (*s1 == *s2)) {
+        if(--len == 0) {
+            return 0;
+        }
+        s1++;
+        s2++;
+    }
+    return *(const unsigned char *)s1 - *(const unsigned char *)s2;
```

Comment:
Handle the case where `len` is `0`. Neither pointer is safe to dereference in that case.

Reply:
Good catch. Will fix that now.